Don't use Profiler key in lite interpreter

ilia-cher opened the pull request:
Summary:
Temporarily re-enable RecordFunction in lite interpreter when profiler key is not set,
this allows the profiler to work without profiled wrappers in the build

Test Plan: CI

Differential Revision: D21409120

smessmer commented on torch/csrc/jit/mobile/interpreter.cpp:
```diff
@@ -43,8 +45,17 @@ bool InterpreterState::run(Stack& stack) {
-        RECORD_FUNCTION(code_->op_names_[inst.X].name, stack);
 #endif
+        //TODO(iliacher): remove the workaround after RecordFunction is in Dispatcher
+        bool prev_value = observersEnabled();
```
nit: this could be simpler by writing

```cpp
bool prev_value = observersEnabled();
enableObservers(true);
RECORD_FUNCTION(code_->op_names_[inst.X].name, stack);
enableObservers(prev_value);
```

but the current way is fine too, up to you.
Also, I'm not sure how this value is stored. If `enableObservers` modifies a non-thread-local or an atomic variable, then your approach is better because it avoids modifying that.

ilia-cher replied:
I wanted to avoid the unnecessary calls, enableObservers sets dispatcher TLS key (at the moment, when we move RecordFunction in dispatcher it will set a separate TLS flag)

Commit:
Don't use Profiler key in lite interpreter (#37962)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/37962

Temporarily re-enable RecordFunction in lite interpreter when profiler key is not set,
this allows the profiler to work without profiled wrappers in the build

Test Plan: CI

Reviewed By: smessmer, linbinyu

Differential Revision: D21409120

fbshipit-source-id: e1479bfc8d308372d591488c9c7ea645774cf5be
